ArmRobot/Assets/Scripts/RandomizerPositionObject.cs

```diff
@@ -0,0 +1,165 @@
+    {
+        /*  Picks a 2D point randomly at uniform. Must be between minRadius and
+         *  maxRadius from center and not at the same place than the cube. 
+         *  Point given relative to center.      
+         */
+        //for (var j = 0; j < 100; j++)
+        while (true)
+        {
+            // pick a random point between the square of edge 2*minRadius and the one with edge 2*maxRadius to be sure that the point is reachable by the robot 
+            float randomX = 0.25f;
+            float randomZ = 0.25f;
+            float randomSignX = Random.value;
+            float criticalValueForSign = 0.5f;
+            if (randomSignX > criticalValueForSign){
+                randomX = minRadius + (maxRadius - minRadius) * Random.value;
+            }
+            else {
+                randomX = -maxRadius + (-minRadius - (-maxRadius)) * Random.value;
+            }
+
+            float randomSignZ = Random.value;
+            if (randomSignZ > criticalValueForSign){
+                randomZ = minRadius + (maxRadius - minRadius) * Random.value;
+            }
+            else {
+                randomZ = -maxRadius + (-minRadius - (-maxRadius)) * Random.value;
+            }
+            Vector2 randomPoint = new Vector2(randomX, randomZ); // now we are sure the point is in the area reachable by the robot 
+
+            // now we need to check if their is no conflict with the position of objects already moved 
+            // we define a flag that will give information on whether or not the random point fits 
+            bool flag = true; 
+            
+            // we iterate through all the list of already moved objects 
+            for (var i = 0; i < listOfAlreadyMovedObjects.Count; i++){
+                gameobjectSeen = listOfAlreadyMovedObjects[i];
+
+                // first we will compute the distance between the datapoint generated and one of the object of the list of already moved objects 
+                Vector2 inter_object = new Vector2(gameobjectSeen.transform.position.x - randomX, gameobjectSeen.transform.position.z - randomZ);
+                float distance_between_objects = inter_object.magnitude;
+                
+                // now we need to access to the radius of the gameobjectSeen
+                float gameobjectSeenRadius = 0f; 
+                if (gameobjectSeen.tag == "Sphere"){
+                    gameobjectSeenRadius = gameobjectSeen.GetComponent<SphereCollider>().radius * gameobjectSeen.GetComponent<Transform>().localScale[0]; // for the sphere 
+                }
+                else if (gameobjectSeen.name == "Cube"){
+                    gameobjectSeenRadius = Mathf.Sqrt(2) * gameobjectSeen.GetComponent<BoxCollider>().bounds.extents.x;
+                }
+                else{
+                    gameobjectSeenRadius = gameobjectSeen.GetComponent<CapsuleCollider>().radius * gameobjectSeen.GetComponent<Transform>().localScale[0]; // for the cylinder
+                }
```

Comment:
This section of code could be moved into its own method to improve readability - maybe `GameObjectRadius(GameObject gameObject){}` or something like that.